flytekit/core/data_persistence.py

```diff
@@ -136,13 +138,20 @@ def register_plugin(cls, protocol: str, plugin: typing.Type[DataPersistence], fo
 
         cls._PLUGINS[protocol] = plugin
 
+    @staticmethod
+    def get_protocol(url: str):
+        parts = re.split(r"(\:\:|\://)", url, 1)
+        if len(parts) > 1:
+            return parts[0]
+        return "file"
```

Comment:
sorry for being dense, but how does this work with windows paths?

Reply:
it will not generate a protocol like `c:\\` if path is `c:\\app\\test`. The part will become `["c:\\app\\test"]` instead, and will use dick persistence.

Reply:
ok, I see it, thanks for explaining.

Why can't we import `get_protocol` from `fsspec` directly?

Reply:
we don't have the fsspec package in flytekit dependency now.